docs/walkthrough.md

```diff
@@ -0,0 +1,500 @@
+    regex: (.+)
+```
+
+</details>
+
+Place your full configuration (it might just be the above) in a file (call
+it `prom-cfg.yaml`, for example), and then create a ConfigMap containing
+it:
+
+```shell
+$ kubectl create configmap prometheus --from-file=prometheus.yml=prom-cfg.yaml
+```
+
+You'll be using this later when you launch Prometheus.
+
+### Launching the Deployment ###
+
+It's generally easiest to launch the adapter and Prometheus as two
+containers in the same pod.  You can use a deployment to manage your
+adapter and Prometheus instance.
+
+Start out with a fairly straightforward Prometheus deployment using the
+ConfigMap from above, and proceed from there:
+
+<details>
+
+<summary>prom-adapter.deployment.yaml [Prometheus only]</summary>
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: prometheus
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus
+  template:
+    metadata:
+      labels:
+        app: prometheus
+    spec:
+      containers:
+      - image: prom/prometheus:v1.6.1
+        name: prometheus
+        args:
+        - -storage.local.retention=6h
+        - -storage.local.memory-chunks=500000
+        # point prometheus at the configuration that you mount in below
+        - -config.file=/etc/prometheus/prometheus.yml
+        ports:
+        # this port exposes the dashboard and the HTTP API
+        - containerPort: 9090
+          name: prom-web
+          protocol: TCP
+        volumeMounts:
+        # you'll mount your ConfigMap volume at /etc/prometheus
+        - mountPath: /etc/prometheus
+          name: prom-config
+      volumes:
+      # make your configmap available as a volume, so that you can
+      # mount in the Prometheus config from earlier
+      - name: config-volume
+        configMap:
+          name: prometheus
+```
+
+</details>
+
+Save this file (for example, as `prom-adapter.deployment.yaml`).  Now,
+you'll need to modify the deployment to include the adapter.
+
+The adapter has several options, most of which it shares with any other
+standard Kubernetes addon API server.  This means that you'll need proper
+API server certificates for it to function properly.  To learn more about
+which certificates are needed, and what they mean, see [Concepts: Auth and
+Certificates](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/concepts/auth.md).
+
+Once you've generated the necessary certificates, you'll need to place
+them in a secret.  This walkthrough assumes that you've set up your
+cluster to automatically inject client certificates and CA certificates
+(see the above concepts documentation).  Make sure you've granted the
+default service account for your namespace permission to fetch the
+authentication CA ConfigMap:
+
+```shell
+$ kubectl create rolebinding prom-ext-auth-reader --role="extension-apiserver-authentication-reader" --serviceaccount=prom:default
+```
+
+Then, store your serving certificates in a secret:
```

Comment:
I've linked to a document that provides more details on how to do it (the "Concepts" doc on auth and certs).  I don't want to have that info reproduced everywhere, because it makes it harder to update.